tests/unit/db/account_tests.py

```diff
@@ -0,0 +1,640 @@
+        resp = requests.delete(
+            'http://{0}:{1}@{2}/{3}'.format(user, pwd, admin_ep, user)
+            )
+        resp.raise_for_status()
+
+    def setUp(self):
+        """
+        Set up test attributes for CouchDB Account tests
+        """
+        self.user = os.environ.get('COUCHDB_USER', 'unittest')
+        self.pwd = os.environ.get('COUCHDB_PASSWORD', 'unittestpassword')
+        self.url = 'http://{0}'.format(
+            os.environ.get('COUCHDB_HOST', '127.0.0.1:5984')
+            )
+        self.client = CouchDB(self.user, self.pwd, url=self.url)
+
+    def tearDown(self):
+        """
+        Ensure the client is new for each test
+        """
+        del self.client
+
+    def dbname(self, database_name='account-unit-test-db'):
+        return '{0}-{1}'.format(database_name, unicode(uuid.uuid4()))
+
+    def test_constructor_with_url(self):
+        """
+        Test instantiating an account object using a URL
+        """
+        self.assertEqual(
+            self.client.cloudant_url,
+            self.url
+            )
+        self.assertEqual(self.client.encoder, json.JSONEncoder)
+        self.assertIsNone(self.client.r_session)
+
+    def test_connect(self):
+        """
+        Test connect and disconnect functionality
+        """
+        try:
+            self.client.connect()
+            self.assertIsInstance(self.client.r_session, requests.Session)
+            self.assertEqual(self.client.r_session.auth, (self.user, self.pwd))
+        finally:
+            self.client.disconnect()
+            self.assertIsNone(self.client.r_session)
+
+    def test_session(self):
+        """
+        Test getting session information
+        """
+        try:
+            self.client.connect()
+            session = self.client.session()
+            self.assertEqual(session['userCtx']['name'], self.user)
+        finally:
+            self.client.disconnect()
+
+    def test_session_cookie(self):
+        """
+        Test getting the session cookie
+        """
+        try:
+            self.client.connect()
+            self.assertIsNotNone(self.client.session_cookie())
+        finally:
+            self.client.disconnect()
+
+    def test_basic_auth_str(self):
+        """
+        Test getting the basic authentication string
+        """
+        try:
+            self.client.connect()
+            expected = 'Basic {0}'.format(
+                base64.urlsafe_b64encode("{0}:{1}".format(self.user, self.pwd))
+                )
+            self.assertEqual(
+                self.client.basic_auth_str(),
+                expected
+                )
+        finally:
+            self.client.disconnect()
+
+    def test_all_dbs(self):
+        """
+        Test getting a list of all of the databases in the account
+        """
+        try:
+            self.client.connect()
+            self.assertIsInstance(self.client.all_dbs(), list)
```

Comment:
Changed to create a set of databases and to check for those databases in the returned list from the call to `all_db()`.  Done in d2f7c4a.